extractors/src/main/scala/sqlest/extractor/Extractor.scala

```diff
@@ -38,6 +38,23 @@ sealed trait Extractor[Row, A] {
 
   def map[B](func: A => B) = MappedExtractor(this, func)
   def map[B](func: A => B, unapplyFunc: B => Option[Any]) = MappedExtractor(this, func, Some(unapplyFunc))
+
+  def cond[B](choices: (A => Boolean, Extractor[Row, B])*): Extractor[Row, B] = {
+    if (choices.isEmpty)
+      new ExceptionExtractor(throw new MatchError("ChoiceExtractor has no matching case"))
+    else {
+      val (pred, extractor) = choices.head
+      val rest = choices.tail
+      choose(pred)(extractor, cond(rest: _*))
+    }
+  }
```

Comment:
This might be better if it was generated using sbt-boilerplate, e.g.
```scala
def cond[B1 <: C, C](choice1: (A => Boolean, Extractor[Row, B1]): Extractor[Row, C] = {
  val (pred, extractor) = choice1
  choose(pred)(extractor, ExceptionExtractor("ChoiceExtractor has no matching case")))
}

def cond[B1 <: C, B2 <: C, C](choice1: (A => Boolean, Extractor[Row, B1], choice2: (A => Boolean, Extractor[Row, B2]): Extractor[Row, C] = {
  val (pred, extractor) = choice1
  choose(pred)(extractor, cond(choice2))
}
```